Bugfix in eltwise ref for fsv4 layout

byungilm opened the pull request:
+ Bugfix in CheckInputsOutputNoPitchSameDims

Signed-off-by: Min, Byungil <byungil.min@intel.com>

### Details:
 - *item1*
 - *...*

### Tickets:
 - *ticket-id*

popovaan committed:
Roll reference implementation (#4947)

* Added reference implementation for Roll operation.

* Small corrections.

* Removed duplicate test disabling.

* Changed implementation using manual data manipulation.

* Removed unnecessary function.

* Corrected tests, added converting axes and shift to int64.

Commit:
Add more features to span and unit tests (#5002)

* Add more span util and test

* apply review suggestions

* add throw in at operator in span

Co-authored-by: Patryk Elszkowski <patryk.elszkowki@intel.com>

Commit:
[IE Tools]  replace .format on f-string (#4970)

* replace .format on f-string in cross_check_tool

* Replace f-string on .format in utils.py

* replace f-string in benchmark tool

* Replace .format on f-string in benchmark tool

* Add f-string after update

* Fix some lines

* Fix utils

Commit:
Add get_function to ONNX Editor API (#5045)

iefode committed:
[IE TESTS] Fix conformance test names (#5095)

* [IE TESTS] Fix conformance test names

* sdl;

* APPLY COMMENTS FROM MIKHAIL

tsavina committed:
Add POT how-to (#5092)

Commit:
setupvars and demos: Fix getting script dir on macOS 10.15 with with default zsh shell (#5094)

byungilm committed:
Bugfix in eltwise ref for fsv4 layout

+ Bugfix in CheckInputsOutputNoPitchSameDims
+ Bugfix in eltwise_vload8
+ Add testcases

Signed-off-by: Min, Byungil <byungil.min@intel.com>

bszmelcz committed:
Revise mod (#5060)

* clean the FusedOp from mod operation

* add backend and type_prop tests for mod operator

* convert taking autobrodcast to match binary elementwise arithmetic ops

* add type_prop/mod.cpp to CMakeLists.txt

* fix style

* fix style v2

* remove evaluate method and add backend test for negative numbers

* add copyright for type_prop/mod.cpp

mitruska committed:
DefromablePSROIPooling shape inference update (#4897)

* Test dynamic interval batch dim propagation

* Fix interval batch dim propagation

* Additional shape and attributes checks

* Update type_prop tests

* Comment update

* Update ngraph_reader test

* Revert group_size dims check

* Update class docs

* Fix mode init value

* Add new ngraph reader test with correct pooling mode

* Comments update

* Update error messages

* Update rank checks to use compatible

* Add newline at EOF

* Update ngraphReader test names

yunji-yunji committed:
[IE CLDNN] lower the tolerance of scatter elements update fusing test from 1 to 1e-5 (#4953)

Commit:
[ Transpose sinking ] Transpose->FQ->Reduce (#5026)

* Utils: make_try_fold, clone_try_fold. Template node creation and attempt to fold it

* RTTI for ArithmeticReduction(KeepDims)

* Enriched ngraph::get_default_order overloads with ones for dynamic shape and rank

* [ Transpose sinking ] Transpose->FQ->Reduce to FQ->Reduce->Transpose

* Style: deleted empty line

* RTTI in Reduction operations

* RTTI for LogicalReductionKeepDims

* Transpose: optimizations moved from algebraic simplification to TransposeSinking

* renamed file

* Fix test

* keep_dims is always initialized

* Apply suggestions from code review

Co-authored-by: Gleb Kazantaev <gleb.nnstu@gmail.com>

Co-authored-by: Gleb Kazantaev <gleb.nnstu@gmail.com>

iefode committed:
Update merge_xmls.py (#5103)

vzinovie committed:
[LPT] Mixed precision GPU Plugin tests (#4498)

Commit:
* SIMD opt for the Resize 32F1C (#5025)

kelvinchoi-intel committed:
[IE CLDNN] ScatterNDUpdate: Fix Klocwork issue (#5027)

Signed-off-by: Kelvin Choi <kelvin.choi@intel.com>

lazarevevgeny committed:
Fixed extracting TF models with string constants (#5110)

* Fixed extracting TF models with string constants

* Fixed parsing strings from the TF models

slyubimt committed:
Add entry point for model optimizer as 'mo' alias (#5052)

* Add entry point for model optimizer as 'mo' alias

* Add entry point for mo package

sshlyapn commented on inference-engine/thirdparty/clDNN/kernel_selector/core/common/kernel_selector_utils.cpp:
```diff
@@ -220,35 +220,44 @@ std::vector<size_t> GetOptimalLocalWorkGroupSizes(std::vector<size_t> gws, const
 bool CheckInputsOutputNoPitchSameDims(const base_params& params) {
     bool no_pitch_same_dims = true;
 
+    std::vector<std::pair <DataLayout, int>> block_layouts = {
+        {DataLayout::b_fs_yx_fsv16,          16},
+        {DataLayout::b_fs_zyx_fsv16,         16},
+        {DataLayout::b_fs_yx_fsv32,          32},
+        {DataLayout::b_fs_zyx_fsv32,         32},
+        {DataLayout::bs_fs_yx_bsv16_fsv16,   16},
+        {DataLayout::bs_fs_zyx_bsv16_fsv16,  16},
+        {DataLayout::bs_f_bsv8__af8,         8},
+        {DataLayout::bs_f_bsv16__af8,        16},
+        {DataLayout::b_fs_yx_fsv4,           4},
+        {DataLayout::fs_b_yx_fsv32,          32},
+        {DataLayout::b_fs_yx_32fp,           32}
+    };
```
Looks like it is better to use std::map for this purpose
BTW what about double blocked format e.g. bs_fs_yx_bsv16_fsv16?

sshlyapn commented on inference-engine/thirdparty/clDNN/kernel_selector/core/common/kernel_selector_utils.cpp:
```diff
@@ -220,35 +220,44 @@ std::vector<size_t> GetOptimalLocalWorkGroupSizes(std::vector<size_t> gws, const
+                    for (size_t in = 0; in < fused_op.tensors.size(); in++) {
+                        if (fused_op.tensors[in].LogicalSize() == 1)
+                            continue;
+                        if (fused_op.tensors[in].GetLayout() == layout.first && fused_op.tensors[in].Feature().v % layout.second != 0)
+                            return false;
```
Is this check really necessary? Can this be covered by `PitchesDifferFromLogicalDims()` call?

v-Golubev committed:
[LPT] Concat*Transformation: handled FQ with unexpected quantization levels (#5111)

* [LPT] TRANSFORMATION_EXCEPTION segfault fix

* [LPT] Concat*Transformation: handled FQ with unexpected quantization levels

* [TESTS] Concat*Transformation: added test-cases with unexpected quant levels

iimironov committed:
Update docs for wide and deep model convertion (#5074)

* Update docs for wide and deep model convertion

* Update wide and deep doc

egerashc committed:
[IE CLDNN] Improved undef generation (#4762)

ggalieroc committed:
Convolution group operations alignment (#4387)

* Convolution: Enhance dynamic shape inference of validate and infer types method

* Convolution: Change onnx test with dynamic shapes to float element type

* Convolution: Remove test instances with integer precision

* Convolution: Add backticks to types in spec

* Convolution: Change element type variable for output element type

* GroupConvolution: Add backticks to types in spec

* GroupConvolution: Enhance dynamic shape inference of validate and infer types method

* GroupConvolution: Remove serialization test instances with integer precision

* GroupConvolutionBackpropData: Remove serialization test instances with integer precision

* GroupConvolutionBackpropData: Enhance dynamic shape inference of validate and infer types method

* Convolution: Add helper function to validate convolution parameters in ref impl

* Convolution: Rewrite lambda to capture spatial dims of filters in validate and infer types

* GroupConvolution: Refactor reference implementation

* Remove call to old implementation of convolution using dilations
* Added validation method to validate shapes

* GroupConvolutionBackpropData: Add more type_prop unit test and refactor test names

* Convolution: Extended validation of convolution parameters in reference implementation

* GroupConvolution: Extended validation of group convolution parameters in reference implementation

* GroupConvolutionBackpropData: Add helper function to validate convolution backprop parameters in ref impl

* Clean up unnecessary lines

* BinaryConvolution: Use validate helper function from convolution ref impl

* Convolution: Refactor validate and infer types to improve readability

* BinaryConvolution: Refactor validate and infer types to improve readability

* Convolution: Add explicit tensor shape dims for inputs and outputs in spec

* BinaryConvolution: Add explicit tensor shape dims for inputs and outputs in spec

* GroupConvolution: Add explicit tensor shape dims for inputs and outputs in spec

* Add helper function to infer convolution forward output shape

* Convolution: Refactor validate and infer types to use helpers to infer output shape

* BinaryConvolution: Refactor validate and infer types to use helpers to infer output shape

* GroupConvolutionBackpropData: Fix formula to calculate output shape in validation functions

* Remove symbol to export convolution output shape inference function

* GroupConvolution: Add validation checks for input channels dim of data batch and filter shape

* GroupConvolutionBackpropData: clean up type prop tests

* Convolution: Change element type in onnx unit tests with dyn shapes and convolution nodes

* GroupConvolutionBackpropData: Correct layout of filters input

* GroupConvolution: Deduce groups from inputs shape during output shape inference

* Change spec supported types of convolution operations to any numeric type

* Revert "GroupConvolution: Remove serialization test instances with integer precision"

This reverts commit 781c2570d6991ef8d1a0ca6d56f00d4ce2fb2a8a.

* Revert "GroupConvolutionBackpropData: Remove serialization test instances with integer precision"

This reverts commit 9a6ac23968befe828eaae13918dac9ce2f2f1704.

* Revert "Convolution: Remove test instances with integer precision"

This reverts commit 0b07052a621135d44242985614bf628ae046d7b3.

* Revert "Convolution: Change element type in onnx unit tests with dyn shapes and convolution nodes"

This reverts commit c9f5944b6bc956d63df28b3fd6d1b19160f57bdd.

* Revert "Convolution: Change onnx test with dynamic shapes to float element type"

This reverts commit 1f4202b010883fa721a3b3a80f57bf4e09f4e697.

* Allow integral types in validate and infer types method for convolution group of operations

* Add i32 precision in single layer tests for convolution group of operations

* BinaryConvolution: Fix shape of input and output tensors in spec

* Address nitpick comments

Commit:
Update specifications for experimental opeartions (#5085)

byungilm committed:
Fix comment

Signed-off-by: Min, Byungil <byungil.min@intel.com>

hyunback committed:
[IE CLDNN] Support broadcasting in etlwise b_fs_yx_fsv16 (#4996)

v-Golubev committed:
[LPT] Split & VariadicSplit support (#4195)

* [LPT] Split support in ConcatTransformation

* [LPT] fixing functional problems after enabling Split/VariadicSplit transformations

* [LPT] added test case for StridedSliceTransformation, enabled tests with split

* [LPT] ConcatTransformation refactoring

* ConcatTransformation: added axis check

* [LPT] Added foldDequantizationConstant to NetworkHelper & SplitTransformation refactoring

* [LPT] Subgraph: returned and refactored quantizationPerChannel

* [LPT] foldDequantizationConstant refactoring

* [LPT] SplitTransformation refactoring

* [LPT] hbonets fix & ConcatTrasnformation refactoring

Commit:
Revert "[LPT] Split & VariadicSplit support (#4195)" (#5129)

This reverts commit 7877287301748c0cd6f93d2813a714fa37824b11.

mateusztabaka committed:
[ONNX] remove ReduceMax and Subtract from LogSoftmax (#5120)

Commit:
Introduced cmake flag to control ONNX Editor build (#5081)

sirzabek committed:
GNA low precision (#4109)

* [GNA] Add low precision support for some layers (affine, eltwise, ReLU, sigmoid)

* [GNA] Fix convolution and scaleshift regression fail

* [GNA] Fix const scale factor calc

* [GNA] Fix extra whitespace

* [GNA] Eltwise padding and low precision primitive creation support

* [GNA] Changes after review

* [GNA] Remove INPUT_PRECISION GNA flag

Remove INPUT_PRECISION flag, all references to it
and tests using it. Other minor stylistic changes and cleanup.

anton-potapov committed:
[PP] Support for FP16 in Reorder cases (#4427)

- added basic support for FP16 (plain wrapper over int16_t)
 - extended Split/Merge operations to support it
 - tests

a-sidorova committed:
[CPU] Added GELU-7 support (#4800)

nosovmik committed:
[TimeTests] Add test case for testing first time inference with caching enabled (#4933)

Important note: Invoker (e.g. test_timetest.py) must ignore first measurement when cache is not generated

mateusztabaka committed:
Disable AddFakeQuantizeFusion and MulFakeQuantizeFusion (#5127)

v-Golubev committed:
[LPT] Split/VariadicSplit support (#5132)

* [LPT] Split support in ConcatTransformation

* [LPT] fixing functional problems after enabling Split/VariadicSplit transformations

* [LPT] added test case for StridedSliceTransformation, enabled tests with split

* [LPT] ConcatTransformation refactoring

* ConcatTransformation: added axis check

* [LPT] Added foldDequantizationConstant to NetworkHelper & SplitTransformation refactoring

* [LPT] Subgraph: returned and refactored quantizationPerChannel

* [LPT] foldDequantizationConstant refactoring

* [LPT] SplitTransformation refactoring

* [LPT] hbonets fix & ConcatTrasnformation refactoring

* [LPT] ConcatTransformation functional tests quick fix

Commit:
[IE CLDNN] Add sub-device support to GPU plugin (#4984)

sadolini committed:
fix bug: assert is keyword, not function (#5131)

* fix bug: assert is keyword, not function

* fix comment

apankratovantonp committed:
Used internal exec network impl (#4922)

dpigasin committed:
Update IE Python Samples (#4907)

* refactor: update ie python samples

* python samples: change comment about infer request creation (step 5)

* python sample: add the ability to run object_detection_sample_ssd.py with a model with 2 outputs

* Add batch size usage to python style transfer sample

* Change comment about model reading

* Add output queue to classification async sample

* add reshape for output to catch results with more than 2 dimensions (classification samples)

* Set a log output stream to stdout to pass the hello query device test

* Add comments to the hello query device sample

* Set sys.stdout as a logging stream for all python IE samples

* Add batch size usage to ngraph_function_creation_sample

* Return the ability to read an image from a ubyte file

* Add few comments and function docstrings

* Restore IE python classification samples output

* Add --original_size arg for python style transfer sample

* Change log message to pass tests (object detection ie python sample)

* Return python shebang

* Add comment about a probs array sorting using np.argsort

a-sidorova committed:
[CPU] Added improvements for StridedSlice (#4248)

Commit:
Added more comments into speech sample (#5049)

yekruglov committed:
[ MO ] Warnings fix (#5105)

* warnings fix

* remove redundant backslash

* add whitespace in the error message

Commit:
update OpenCV version to 4.5.2 (#5069) (#5134)

* update OpenCV version to 4.5.2

mryzhov committed:
Corrected versions of numpy and pandas (#5106)

Aligned versions with the latest POT changes

mryzhov committed:
Revert "Update IE Python Samples (#4907)" (#5151)

This reverts commit 003474937dfa7cb1cd22b2bc1fbc2f39ad5eeb69.

popovaan committed:
Fixed framework name attribute in mapping file. (#5046)

* Fixed framework name attribute for onnx, mxnet.

* Fixed framework name attribute for caffe.

* Removed unnecessary attribute setting from add_opoutput()

* Added identity nodes adding to outputs in mxnet loader.

* Removed unnecessary reformat.

* Removed unnecessary reformat.

* Added check for empty name.

* Used nodes indices instead of node names in loader.

* Code refactoring, small bug fixed.

byungilm committed:
Apply comment

Signed-off-by: byungilm <byungil.min@intel.com>

KorA6 committed:
[IE][VPU]: firmware update (#5143)

* Updating avgPooling errors tracking.

byungilm committed:
[IE CLDNN] Resample opt caffe (#4770)

EgorDuplensky committed:
Correct benchmark_app description of enforcebf16 parameter (#4958)

* Correct benchmark_app description of enforcebf16 parameter

* Align enforcebf16 parameter description across cpp and py benchmark_apps

* Fix typo

* Fix another typo

byungilm committed:
Bugfix

Signed-off-by: Min, Byungil <byungil.min@intel.com>
